### PR TITLE
test(client): add tests for buildNFTAnnotations

### PR DESCRIPTION
**Pull request by aqrln**
https://github.com/prisma/prisma/pull/9371#issuecomment-1046882767

### CODE REVIEW

**Comment by Jolg42** [comment]
Looks like tests are failing on Windows 
```
    - Snapshot  - 2
    + Received  + 2

      ↵
      path.join(__dirname, "libquery_engine-TEST_PLATFORM.LIBRARY_TYPE.node");
    - path.join(process.cwd(), "out/libquery_engine-TEST_PLATFORM.so.node")
    + path.join(process.cwd(), "out//libquery_engine-TEST_PLATFORM.so.node")
      path.join(__dirname, "schema.prisma");
    - path.join(process.cwd(), "out/schema.prisma")
    + path.join(process.cwd(), "out//schema.prisma")
```

The double slash `//` doesn't look good we should check and understand what's going on here, and potentially revert the original PR @aqrln @millsp

**Comment by aqrln** [comment]
@Jolg42 

> The double slash // doesn't look good we should check and understand what's going on here, and potentially revert the original PR @aqrln @millsp

It's because of our snapshot serializer, it always replaces backslashes with forward slashes on Windows, sometimes even outside of file names (some tests are skipped on Windows because of it).

That said, if the snapshot serializer didn't replace them here, the snapshots would still fail because of double backslashes.

File: packages/client/src/__tests__/buildNFTAnnotations.test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    // TODO: when using .toMatchInlineSnapshot(), this fails after updating snapshots.
+    // Probably an issue with the snapshot serializer?
+    expect(normalizePaths(annotations)).toBe(`
+
+`)
+  })
+})
+
+describe('special cases', () => {
+  // TODO: document the reason behind this logic.
```
Comment (Jolg42) [inline packages/client/src/__tests__/buildNFTAnnotations.test.ts:101]: From what I remember, the build image is different from the runtime image, so a different engine is needed.
Not sure what's the current state as of today.

Comment (janpio) [reply packages/client/src/__tests__/buildNFTAnnotations.test.ts:101]: Yep, build process was local (debian) but deployment was Lamba (rhel) which lead to people having to do weird things - or us just reacting to `NETLIFY=true` and replacing/adding/removing things to make sure.

**Comment by aqrln** [comment]
Rebased and added a comment, please take another look.

File: packages/client/src/generation/utils/buildNFTAnnotations.ts [hunk 2]
```diff
@@ -49,6 +51,7 @@ function getQueryEngineFilename(engineType: ClientEngineType, platform: Platform
   }
 
   if (engineType === ClientEngineType.Binary) {
+    // TODO: is this correct on Windows?
```
[the remark this answers is not in the document]
Comment (aqrln) [reply packages/client/src/generation/utils/buildNFTAnnotations.ts:54]: basically yes

[the remark this answers is not in the document]
Comment (Jolg42) [reply packages/client/src/generation/utils/buildNFTAnnotations.ts:54]: @millsp I think for annotations the full path is not needed? Only the beginning is enough?

[the remark this answers is not in the document]
Comment (millsp) [reply packages/client/src/generation/utils/buildNFTAnnotations.ts:54]: Yep, I don't think it matters.

[the remark this answers is not in the document]
Comment (aqrln) [reply packages/client/src/generation/utils/buildNFTAnnotations.ts:54]: cool, I'll remove the comment then

**Review by millsp — APPROVED** [review]

**Pull request merged**